src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -269,16 +280,117 @@ describe('PaymentButton', () => {
+
+    // - Spinner should be displayed
+    screen.getByText('Payment in progress');
+
+    // - Payment interface should not be displayed
+    expect(screen.queryByText('Payment interface component')).toBeNull();
+
+    // - Order should be polled until its state is validated
+    fetchMock.get(
+      `https://joanie.test/api/v1.0/orders/${order.id}/`,
+      {
+        ...order,
+        state: OrderState.VALIDATED,
+      },
+      {
+        overwriteRoutes: true,
+      },
+    );
+
+    // - Advance timer to one tick
+    await act(async () => {
+      jest.runOnlyPendingTimers();
+    });
+
+    // - Order should have been refetched
+    expect(fetchMock.calls()).toHaveLength(4);
+    expect(fetchMock.lastUrl()).toBe(`https://joanie.test/api/v1.0/orders/${order.id}/`);
+
+    // - As order state is validated, the onSuccess callback should be triggered.
+    expect(handleSuccess).toHaveBeenCalledTimes(1);
+    //
+    // - And poller should be stopped
+    await act(async () => {
+      jest.runOnlyPendingTimers();
+    });
+    expect(fetchMock.calls()).toHaveLength(4);
+  });
+
+  it('should render a payment button and not call the order creation route', async () => {
+    const product: Joanie.Product = ProductFactory().one();
+    const billingAddress: Joanie.Address = AddressFactory().one();
+    const creditCard: Joanie.CreditCard = CreditCardFactory().one();
+    const { payment_info: paymentInfo, ...order }: Joanie.OrderWithPaymentInfo =
+      OrderWithOneClickPaymentFactory().one();
+    const handleSuccess = jest.fn();
+    const initialOrder = {
+      ...order,
+      state: OrderState.PENDING,
+    };
+
+    fetchMock
+      .post('https://joanie.test/api/v1.0/orders/', order)
+      .patch(`https://joanie.test/api/v1.0/orders/${order.id}/submit/`, {
+        payment_info: paymentInfo,
+      })
+      .get(`https://joanie.test/api/v1.0/orders/${order.id}/`, {
+        ...order,
+        state: OrderState.SUBMITTED,
+      });
+
+    render(
+      <Wrapper order={initialOrder} client={createTestQueryClient({ user: null })}>
+        <PaymentButton
+          billingAddress={billingAddress}
+          creditCard={creditCard.id}
+          product={product}
+          onSuccess={handleSuccess}
+        />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay in one click ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // - Payment button should not be disabled.
+    expect($button.disabled).toBe(false);
+
+    // - User clicks on pay button
+    await act(async () => {
+      fireEvent.click($button);
+    });
+
+    // - Route to submit an existing order
+    // - Furthermore, as payment succeeded immediately, order should have been refetched
+    expect(fetchMock.calls()).toHaveLength(2);
+
+    expect(fetchMock.calls()[0][0]).toBe(`https://joanie.test/api/v1.0/orders/${order.id}/submit/`);
```

Comment:
we don't care about api calls order, we can use something like: 
```js
const calledUrls = fetchMock.calls().map((call) => call[0]);
expect(calledUrls).toContain(`https://joanie.test/api/v1.0/orders/${order.id}/submit/`)
```

note: also usefull to test that an url haven't been called with `not.toContain()`

Reply:
Nope, the order is very important in this case !